offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/offergeneration/OfferTemplate.groovy

```diff
@@ -0,0 +1,370 @@
+package life.qbic.portal.offermanager.offergeneration
+
+import life.qbic.business.offers.Currency
+import life.qbic.business.offers.OfferContent
+import life.qbic.business.offers.OfferItem
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+
+import java.text.DateFormat
+import java.text.DecimalFormat
+
+/**
+ * <b>An offer template that can be filled with an OfferContent</b>
+ *
+ * <p>This class represents a html document structure. It provides filled versions of the same document.</p>
+ *
+ * @since 1.1.0
+ */
+class OfferTemplate {
+
+    private Document document
+    protected static final DecimalFormat PERCENTAGE_FORMATTER = new DecimalFormat("#%")
+
+    /**
+     * Creates a new OfferTemplate using the provided document as template document.
+     * <p>Please note: It is important that the template document provides the following css selectors</p>
+     * <ul>
+     *  <li>.currency-symbol</li>
```

Comment:
why do we need the currency symbol here? We agreed that since its a german offer that there will be no other currency than €

Reply:
I was not aware of this decision. This way the offer will only hold € 🤷 since it is set here explicitly